wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -952,8 +952,8 @@ impl PhysicalDeviceProperties {
                 extensions.push(vk::ExtImageRobustnessFn::name());
             }
 
-            // Require `VK_EXT_subgroup_size_control` if the associated feature was requested
-            if requested_features.contains(wgt::Features::SUBGROUP) {
+            // Require `VK_EXT_subgroup_size_control` if it is available
+            if self.supports_extension(vk::ExtSubgroupSizeControlFn::name()) {
```

Comment:
Note to reviewer, this is inside in a `vk_version < 1.3` block.

Reply:
I think this was correct before, this change makes it so that the `VK_EXT_subgroup_size_control` extension is optional even if the user requested the feature.

Reply:
I'm not sure I follow.

The code for adding the varying subgroup size flag checks if the _adapter_ supports it, not if the user asked for it from the device. As such we need to always ask for the extension if it's available to us.

Reply:
Ah, I see, I think we should fix it here then:

https://github.com/gfx-rs/wgpu/blob/08efa72a83d0acdf25177408150e4c87933c1cd7/wgpu-hal/src/vulkan/adapter.rs#L451-L458

It should be `.subgroup_size_control(requested_features.contains(wgt::Features::SUBGROUP))`.

This is how we deal with multiview as well.

Reply:
It wasn't intended that we always use the functionality if it's available.
We should only use it if the user asked for `Features::SUBGROUP`.

Reply:
Alright, I think I can check that, if we store the user's features in the vk DeviceShared in - the problem is https://github.com/cwfitzgerald/wgpu/blob/cdc67ff9222087494000696aa87b92dfb1de6748/wgpu-hal/src/vulkan/device.rs#L792-L794

Reply:
I thought that if we pass `false` to `vk::PhysicalDeviceSubgroupSizeControlFeatures::builder().subgroup_size_control(false)`, `self.shared.private_caps.subgroup_size_control` should also end up being `false`.

Reply:
No, we pull that from the vulkan physical device features, so it's true if the device supports it.

Reply:
I see, actually `PrivateCapabilities` sounds like the wrong place for that field.

Reply:
@teoxoy take a look at this, I think that should work now?

Reply:
Looks good!